cross/toolchain.cmake

```diff
@@ -57,20 +61,25 @@ macro(add_compile_param)
 endmacro()
 
 # Specify link flags
-add_compile_param(CROSS_LINK_FLAGS "--sysroot=${CROSS_ROOTFS}")
-add_compile_param(CROSS_LINK_FLAGS "--gcc-toolchain=${CROSS_ROOTFS}/usr")
-add_compile_param(CROSS_LINK_FLAGS "--target=${TOOLCHAIN}")
-add_compile_param(CROSS_LINK_FLAGS "-fuse-ld=gold")
+add_compile_param(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} --sysroot=${CROSS_ROOTFS}")
+if (CLR_CMAKE_COMPILER STREQUAL "Clang")
+  message(WARNING "hello clang linker settings")
+  add_compile_param(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} --gcc-toolchain=${CROSS_ROOTFS}/usr")
+  add_compile_param(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} --target=${TOOLCHAIN}")
+  message(WARNING ${CROSS_LINK_FLAGS})
```

Comment:
I actually wonder if the WARNING in the message is not causing a trouble here. I would try to remove it from all the message calls you've added.

Reply:
isn't this a i = i + "mystr1" situation?

When I looked at the error log, I didn't see -gcc-toolchain in clang-5 linker line. That made me look here to see what happened to the -gcc-toolchain parameter.

Even inside the same file today, I see some code as:

 add_compile_param(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS}  value"

and other code as:

 add_compile_param(CROSS_LINK_FLAGS "newvalue"

My thought was that the second pattern is destroying the previously assigned CROSS_LINK_FLAGS due to sequencing issue:

add_compile_param(CROSS_LINK_FLAGS "value1"
add_compile_param(CROSS_LINK_FLAGS "value2"

CROSS_LINK_FLAGS=value2

There was a compilation error without my message statement too.

Reply:
The `add_compile_param` macro adds the option to the list of options if it isn't already in the list. I don't see code like `add_compile_param(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} value")`. I think you might have referred to `add_compile_param(CMAKE_EXE_LINKER_FLAGS "${CROSS_LINK_FLAGS}" "TOOLCHAIN_EXE_LINKER_FLAGS")`, but that's a different thing.

Reply:
You are right, I caught the wrong fish. Let me back out these changes.